python/torch_mlir/eager/torch_dispatch.py

```diff
@@ -0,0 +1,353 @@
+                for candidate_signature, candidate_schema in matches:
+                    sig_matches = True
+                    try:
+                        bound_types = candidate_signature.bind(
+                            *arg_types, **kwarg_types
+                        )
+                        for arg_name, arg_type in bound_types.arguments.items():
+                            param = candidate_signature.parameters[arg_name]
+                            sig_matches = sig_matches and type_matches(
+                                param.annotation, arg_type
+                            )
+                    except TypeError as e:
+                        sig_matches = False
+                    if sig_matches:
+                        new_args_and_kwargs = _args_kwargs_to_normalized_args_kwargs(
+                            candidate_signature,
+                            args,
+                            kwargs,
+                            normalize_to_only_use_kwargs,
+                        )
+                        return (
+                            candidate_schema,
+                            new_args_and_kwargs.args,
+                            new_args_and_kwargs.kwargs,
+                        )
+            else:
+                # matched more than one schema despite providing types. Not sure if this is possible since in the reference
+                # implementation the comment here states caller should provide types (optional for the reference impl)
+                # where we require arg types.
+                schema_printouts = "\n".join(str(schema) for schema in matches)
+                raise RuntimeError(
+                    f"Tried to normalize arguments to {torch.typename(target)} but "
+                    f"the schema match was ambiguous! Please provide argument types to "
+                    f"the normalize_arguments() call. Available schemas:\n{schema_printouts}"
+                )
+
+    raise RuntimeError("couldn't normalize args")
+
+
+# shameless SO copy pasta
+def urlify(s):
+    # Remove all non-word characters (everything except numbers and letters)
+    s = re.sub(r"[^\w\s]", "", s)
+    # Replace all runs of whitespace with a single dash
+    s = re.sub(r"\s+", "_", s)
+
+    return s
+
+
+def build_script_function(
+    schema: torch._C.FunctionSchema,
+    args: List[torch._C.Argument],
+    kwargs: Dict[str, Any],
+) -> torch.jit.ScriptFunction:
+    """
+    Build a torch.jit.ScriptFunction that corresponds to the schema.
+    This is necessary so that we can pass a "hydrated" module to ModuleBuilder::importModule.
+    Hydrated meaning with inlined constants with correct values; a gotcha here is that if you don't inline the constants
+    then importModule assume default values (i think?) thereby (potentially) failing to support the op (not sure how this makes sense but native_batch_norm for example doesn't work without train=False)
```

Comment:
I would describe this process better as that the constant values are essentially similar to the shapes of the input tensors. Just as by default we need to naively compile for every shape, we also need to compile for every set of non-tensor inputs. Only very carefully can we erase their values.